winml/api/Windows.AI.MachineLearning.idl

```diff
@@ -334,6 +334,18 @@ namespace Windows.AI.MachineLearning
         TensorKind KeyKind{ get; };
         //! Returns the properties of the map's value.
         ILearningModelFeatureDescriptor  ValueDescriptor{ get; };
+
+        [contract(MachineLearningContract, 4)]
+        {
+            //! allows creation of feature descriptors outside the runtime (immutable)
+            [method_name("Create")] MapFeatureDescriptor(
+                String Name,
+                String Description,
+                Boolean IsRequired,
+                TensorKind KeyKind,
+                ILearningModelFeatureDescriptor ValueDescriptor
+            );
+        }
```

Comment:
do we need to do API review?

Reply:
also can you explain what these new methods are for?

---
In reply to: [347040835](https://github.com/microsoft/onnxruntime/pull/2406#discussion_r347040835) [](ancestors = 347040835)

Reply:
yep.  we will need an API review.  these new methogs allow creation of feature descriptors outside the runtime (the code comment there).   basically before this change you could never create these on their own but had to have the runtime make then inside of models and return them.   this change adds a factory so that other people can create them.   ORT creates them in this PR.  and in the future people like Vision Skills could create then also. it makes them reusable currencies.

---
In reply to: [347040969](https://github.com/microsoft/onnxruntime/pull/2406#discussion_r347040969) [](ancestors = 347040969,347040835)